src/ensembl/src/shared/button/Buttons.test.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import faker from 'faker';
+
+import { PrimaryButton, SecondaryButton } from './Button';
+
+const onClick = jest.fn();
+
+const defaultProps = {
+  onClick
+};
+
+const renderButton = (
+  ButtonComponent: React.FunctionComponent<any>,
+  props: any = defaultProps
+) => mount(<ButtonComponent {...props} />);
+
+describe('PrimaryButton', () => {
+  let wrapper: any;
+  let buttonChildren: any;
+
+  beforeEach(() => {
+    buttonChildren = faker.lorem.words();
+    wrapper = renderButton(PrimaryButton, {
+      ...defaultProps,
+      children: buttonChildren
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('renders a button', () => {
+    expect(wrapper.find('button').length).toEqual(1);
+  });
+
+  test('assigns the primaryButton class to the button', () => {
+    const renderedButton = wrapper.find('button');
+    expect(renderedButton.hasClass('primaryButton')).toBe(true);
+  });
+
+  test('extends own class with external classname', () => {
+    const externalClass = faker.lorem.word();
+    const buttonChildren = faker.lorem.words();
```

Comment:
Do we need to redeclare `buttonChildren` here?

Reply:
We don't. Fixed.